### PR TITLE
Add a tool to check for missing docs redirects

### PR DESCRIPTION
**Pull request by ptgott**
Add a tool to check for URL paths in the Teleport docs site that do not correspond to actual docs pages or redirects. These URLs are most likely links that will 404 if a user follows them. This way, we can be more strategic about the redirects that we configure for the docs site, meaning that we do not need to add redirects for every single docs page that we rename or delete. A later change can add a GitHub Actions workflow to run the program.

### CODE REVIEW

File: docs/check-redirects/check-redirects.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  checkFile(filePath) {
+    const docsPattern = new RegExp(
+      /https:\/\/goteleport.com\/docs\/(ver\/[0-9]+\.x\/)?[\w\/_#-]+/,
+      'gm'
+    );
+    const text = this.fs.readFileSync(filePath, 'utf8');
+    const docsURLs = [...text.matchAll(docsPattern)];
+    if (!docsURLs) {
+      return;
+    }
```
Comment (avatus) [inline docs/check-redirects/check-redirects.js:85]: should explicitly check for length here (docsURLs is an array yes?)

File: docs/check-redirects/check-redirects.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    let result = [];
+    docsURLs.forEach(url => {
+      const docsPath = this.urlToDocsPath(url[0]);
+      const entry = this.fs.statSync(docsPath, {
+        throwIfNoEntry: false,
+      });
```
Comment (avatus) [inline docs/check-redirects/check-redirects.js:91]: this seems like it should be named `missingEntry` ? if the entry is undefined, we do something right? or maybe like, `entryExists` idk. maybe just a comment "if entry exists, we skip"

**Review by avatus — APPROVED** [review]

**Comment by ptgott** [comment]
TODO: I also need to clarify in the README what the `--in` flag should point to.

**Review by avatus — APPROVED** [review]
Looks great! Thanks

**Review by mmcallister — APPROVED** [review]

**Pull request merged**